frontend/components/common/PrerenderedHTML.tsx

```diff
@@ -313,26 +348,44 @@ export const styles: SystemStyleObject = {
          paddingInline: 3,
       },
    },
+};
 
-   '.fa-svg-icon': {
-      svg: {
-         width: '1em',
-         height: '1em',
-         display: 'inline-block',
-
-         path: {
-            fill: 'currentColor',
-         },
-      },
-
-      '&.xs svg': {
-         width: '0.5em',
-         height: '0.5em',
-      },
-
-      '&.sm svg': {
-         width: '0.75em',
-         height: '0.75em',
-      },
-   },
+const commerceStyles: SystemStyleObject = {
+   ...sharedStyles,
 };
+
+export const PrerenderedHTML = chakra(function Prerendered({
+   html,
+   template,
+   className,
+}: {
+   html: string;
+   template: 'troubleshooting' | 'commerce';
+   className?: string;
+}) {
+   useEffect(() => {
+      if (template === 'troubleshooting') {
+         import('lite-youtube-embed');
+      }
+   }, [template]);
+
+   let sx;
+   switch (template) {
+      case 'troubleshooting':
+         sx = troubleshootingStyles;
+         break;
+      case 'commerce':
+         sx = commerceStyles;
+         break;
+      default:
+         sx = {};
+   }
+
+   return (
+      <Box
+         className={`prerendered ${className}`}
+         sx={sx}
```

Comment:
A possibly cleaner way to do this:
```suggestion
         sx={styleMap[styles]}
```

and then, outside the component:
```ts
const styleMap = {
   troubleshooting: troubleshootingStyles,
   commerce: commerceStyles,
}
```

There's no need for a `default` case, because the type on `styles` only allows the two possible strings.

Reply:
Thanks for the tip. I really like how clean this is now: [`c6d3d62`](https://github.com/iFixit/react-commerce/pull/1946/commits/c6d3d62c4b25aa1fbdfcf713799a77555344810c)